sdk/src/azure/platform/az_mosquitto5.c

```diff
@@ -584,6 +584,8 @@ void az_mqtt5_property_string_free(az_mqtt5_property_string* prop_str)
   _az_PRECONDITION_NOT_NULL(prop_str);
 
   free((void*)prop_str->str._internal.ptr);
+  prop_str->str._internal.ptr = NULL;
```

Comment:
Code outside the `az_span*.c` should _never_ access _internal.

Instead, use:

```C
prop_str->str = AZ_SPAN_EMPTY
```

Same goes for free, etc. In that case you must use `az_span_ptr`.  (Don't assume anything about how a span is implemented: anything in _internal as well as any function behavior may change in the future.)

Reply:
Applied those changes, thanks for pointing it out